frameworks/Swift/vapor/Sources/vapor-tfb/main.swift

```diff
@@ -0,0 +1,24 @@
+import Vapor
+
+// Services
+var services = Services.default()
+
+// Router
+let router = EngineRouter.default()
+// Routes
+let json = Message(message: "Hello, world!")
+router.get("json") { req in
+    return json.encode(status: .ok, headers: ["Server": "Vapor"], for: req)
+}
+let plaintextRes = HTTPResponse(status: .ok, headers: ["Server": "Vapor", "Content-Type": "text/plain"], body: "Hello, world!")
```

Comment:
Like with the `json` variable, I'm wondering why this is declared here instead of inline where it's used.  Do you expect that to affect performance?  `HTTPResponse` is also a `struct` type, right?

Reply:
Yes, there's overhead to initializing an HTTPResponse struct. It should be slightly cheaper to copy the struct value. However, this is by no means a pre-serialized HTTP response. It's just an in-memory representation that will be serialized later to a buffer by the HTTPEncoder.

Reply:
Ok, that makes sense.